frontend/src/stateless/index.ts

```diff
@@ -0,0 +1,344 @@
+      }
+    };
+
+    request.onerror = function (event: any) {
+      reject('Error opening the database: ' + (event.target ? event.target.error : ''));
+    };
+  });
+}
+
+/**
+ * Finds a kubeconfig by cluster name.
+ * @param clusterName
+ * @returns A promise that resolves with the kubeconfig, or null if not found.
+ * @throws Error if IndexedDB is not supported.
+ * @throws Error if the kubeconfig is invalid.
+ */
+export function findKubeconfigByClusterName(clusterName: string): Promise<string | null> {
+  return new Promise<string | null>(async (resolve, reject) => {
+    try {
+      let request;
+      if (process.env.NODE_ENV === 'test') {
+        request = indexedDBtest.open('kubeconfigs', 1);
+      } else {
+        request = indexedDB.open('kubeconfigs', 1);
+      }
+
+      request.onupgradeneeded = function (event: any) {
+        const db = event.target ? event.target.result : null;
+        // Create the object store if it doesn't exist
+        if (!db.objectStoreNames.contains('kubeconfigStore')) {
+          db.createObjectStore('kubeconfigStore', { keyPath: 'id', autoIncrement: true });
+        }
+      };
+
+      request.onsuccess = function (event: any) {
+        const db = event.target.result;
+        const transaction = db.transaction(['kubeconfigStore'], 'readonly');
+        const store = transaction.objectStore('kubeconfigStore');
+
+        store.openCursor().onsuccess = function (event: any) {
+          const cursor = event.target.result;
+          if (cursor) {
+            const kubeconfigObject = cursor.value;
+            const kubeconfig = kubeconfigObject.kubeconfig;
+
+            const parsedKubeconfig = jsyaml.load(atob(kubeconfig)) as KubeconfigObject;
+
+            const matchingKubeconfig = parsedKubeconfig.clusters.find(
+              cluster => cluster.name === clusterName
+            );
+
+            if (matchingKubeconfig) {
+              resolve(kubeconfig);
+            } else {
+              cursor.continue();
+            }
+          } else {
+            resolve(null); // No matching kubeconfig found
+          }
+        };
+      };
+
+      request.onerror = function (event: any) {
+        console.error('Error opening the database:', event.target.error);
+        reject(event.target.error);
+      };
+    } catch (error) {
+      reject(error);
+    }
+  });
+}
+
+/**
+ * In the backend we use a unique ID to identify a user. If there is no ID in localStorage
+ * we generate a new one and store it in localStorage. We then combine it with the
+ * cluster name and this headlamp-userId to create a unique ID for a cluster. If we don't
+ * do it then if 2 different users have a cluster with the same name, then the
+ * proxy will be overwritten.
+ * @returns headlamp-userId from localStorage
+ */
+export function getUserIdFromLocalStorage(): string {
+  let headlampUserId = localStorage.getItem('headlamp-userId');
+
+  if (!headlampUserId) {
+    headlampUserId = uuidv4();
```

Comment:
We do not need a secure token per say, we needed a random uuid to store. Which is then combined with cluster name. Since it's random I don't think it would be easy to guess or hack. 

Please correct me if I don't understand your concern regarding this.

Reply:
As discussed, we have now used crypto package here instead of uuid.